### PR TITLE
chore(AmplifyJS): update the doc of Next.js dx

### PR DESCRIPTION
**Pull request by HuiSF**
#### Description of changes:

* Updated the supported Next.js version range callout
* Updated the "Configure Amplify for Server Side Usage" to match the latest DX pattern
  * Path: `/lib/ssr/nextjs/q/platform/js/`
* Updated the "create your application" page, and adding tabs for the section of "install amplify libraries"
  * Path: `/lib/project-setup/create-application/q/platform/js/`

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [x] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [x] JS
- [ ] iOS
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://link.com)` 
            HTML: `<a href="https://link.com">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/pages/lib/ssr/nextjs/q/platform/[platform].mdx [hunk 1]
```diff
@@ -44,27 +44,33 @@ amplify pull
...
 
-To enable Amplify library API usage on the server side, you will need to update
-your `next.config.js`.
+To use Amplify APIs on the server-side of your Next.js app, you will need to
+create a `runWithAmplifyServerContextRunner` function.
 
-Import the Amplify configuration object from the `amplifyconfiguration.json` file
-that is generated by Amplify CLI, and use the `withAmplify` utility functions to
-update your Next.js app configuration.
+You can create an `amplifyServerRunner.ts` file under `src` in your codebase.
```
Comment (renebrandel) [inline src/pages/lib/ssr/nextjs/q/platform/[platform].mdx:50]: Reason for edit: new Next.js apps discourages use of "src/"
```suggestion
You can create an `amplifyServerRunner.ts` file under a `utils` folder in your codebase.
```

File: src/pages/lib/ssr/nextjs/q/platform/[platform].mdx [hunk 2]
```diff
@@ -44,27 +44,33 @@ amplify pull
...
-const { withAmplify } = require('@aws-amplify/adapter-nextjs/with-amplify');
-const config = require('./src/amplifyconfiguration.json');
-
-/** @type {import('next').NextConfig} */
-const nextConfig = {
-  // your Next.js app configuration
-}
+```typescript
+import { createServerRunner } from "@aws-amplify/adapter-nextjs";
+import config from './amplifyconfiguration.json';
```
Comment (renebrandel) [inline src/pages/lib/ssr/nextjs/q/platform/[platform].mdx:60]: uses aliasing
```suggestion
import config from '@/amplifyconfiguration.json';
```

File: src/pages/lib/ssr/nextjs/q/platform/[platform].mdx [hunk 3]
```diff
@@ -118,7 +124,7 @@ export default function ConfigureAmplifyClientSide() {
 ```
 
 `layout.tsx`:
-```typescript
+```jsx
```
Comment (renebrandel) [inline src/pages/lib/ssr/nextjs/q/platform/[platform].mdx:127]: does `tsx` not work?

File: src/pages/lib/ssr/nextjs/q/platform/[platform].mdx [hunk 4]
```diff
@@ -163,9 +169,9 @@ to the incoming requests in the middleware of your Next.js app to protect your
 routes. For example:
 
 ```typescript
-import { runWithAmplifyServerContext } from '@aws-amplify/adapter-nextjs';
 import { fetchAuthSession } from 'aws-amplify/auth/server';
 import { NextRequest, NextResponse } from 'next/server';
+import { runWithAmplifyServerContext } from './amplifyServerRunner';
```
Comment (renebrandel) [inline src/pages/lib/ssr/nextjs/q/platform/[platform].mdx:174]:
```suggestion
import { runWithAmplifyServerContext } from './utils/amplifyServerRunner';
```

File: src/pages/lib/ssr/nextjs/q/platform/[platform].mdx (hunk 3 above)
Comment (HuiSF) [reply src/pages/lib/ssr/nextjs/q/platform/[platform].mdx:127]: `tsx` doesn't apply any syntax highlighting :/

**Comment by renebrandel** [comment]
Closing in favor of #6313

**Pull request closed without merging**